### ISSUE TITLE
Seeing more cases of failure to render >< in the html with spurious SPACES

### ISSUE THREAD

**Issue by Frenchomatic** [ahmedkaludi/accelerated-mobile-pages#2406]
Additional spaces are being put in where this `><` is in the editor or even in a file for rendering the logo. **The response on the forum wasn't really good enough given this is basic html. Nobody would expect to see something like this being rendered - see where it is putting SPACE: 

`<div class="amp-logo">SPACE<a href="https://www.mydomain.com/amp/">SPACE<amp-img src="https://www.mydomain.com/wp-content/uploads/2017/09/logo.png" width="256" height="60" alt="my logo" class="amp-logo i-amphtml-element i-amphtml-layout-responsive i-amphtml-layout-size-defined i-amphtml-layout" layout="responsive"><i-amphtml-sizer style="display: block; padding-top: 23.4375%;"></i-amphtml-sizer><img decoding="async" alt="my logo" src="https://www.mydomain.com/wp-content/uploads/2017/09/logo.png" class="i-amphtml-fill-content i-amphtml-replaced-content"></amp-img>SPACE</a>SPACE</div>`

or 

`<li>SPACE<strong>Bla Bla</strong> stuff</li>`

or 

`<h2>My title <span class="mycolor">is here</span>SPACE</h2>
`

etc. 

And this is after minimisation with the plugin is being used.

**Comment by Frenchomatic** [comment]
I just went and looked at one file logo.php  -  php cannot be written like this or new lines will be put in. 

Here is just one line and it puts in multiple spaces and especially so when the 'if test' fails. 

          `<a href="<?php echo esc_url( $ampforwp_home_url ); ?>"  <?php if($set_rel_to_noamp){ echo ' rel="nofollow"'; } ?>  >`

Here is what logo.php outputs when no minimization is being used

```

                <div class="amp-logo">
                            <a href="https://www.mydomain.com"    >

                <amp-img src="https://www.mydomain.com/wp-content/uploads/2017/07/logo.png" width="157" height="60" alt="some text" class="amp-logo" layout="responsive"></amp-img>

          </a>
         </div>

```

Now in this case it probably isn't a problem for browsers or google but it will be a problem in many other types of circumstance which are easy to imagine.

**Comment by Frenchomatic** [comment]
Modifying Line 5554 in features.php seems to get rid of these damn spaces but I don't know if there is a reason why it was coded this way.

`        $process = preg_replace(array ('/\>[^\S ]+' . $mod, '/[^\S ]+\<' . $mod, '/\s+/' ), array('> ', ' <', ' '), $process);
`

If I remove the spaces in  `array('> ', ' <', ' '),`  like this then spaces in the html then code like `<li><strong>` renders correctly and so does <h2>bla bla <span class="mycolor">some text</span>. The problem is know what the consequences of doing that are. 


`        $process = preg_replace(array ('/\>[^\S ]+' . $mod, '/[^\S ]+\<' . $mod, '/\s+/' ), array('>', '<', ' '), $process);
`

**Comment by Frenchomatic** [comment]
This modification still doesn't solve the problem with logo.php which has spaces being put in.

**Comment by Frenchomatic** [comment]
So I have an email from you that my edit has no bad side effects  - will this edit be included in the next release?

**Comment by Frenchomatic** [comment]
Answer no it wasn't put into version Version 0.9.97.9

**Comment by merajali294** [comment]
@Zabi09, we need to fix this issue.

**Comment by Frenchomatic** [comment]
see this #2415

**Comment by Zabi09** [comment]
This spaces issues will be fixed in #2445
